File: math/7.c
//注意转换关系 ret = ret * 10 + x % 10

int reverse(int x) {
    long ret = 0;
    while(x != 0)
    {
        ret = ret * 10 + x % 10;
        x = x / 10;
        if(ret < INT_MIN || ret > INT_MAX) return 0;
    }
    return ret;
}

File: math/7.go
import "math"

func reverse(x int) int {
    var flag bool
    var ret int
    if x < 0 {
        flag = true
        x = -x
    }
    for x > 0 {
        ret = ret * 10
        ret += x % 10
        x = x / 10
    }
    
    if flag {
        ret = -ret
    }
    if ret > math.MaxInt32 {
        return 0
    }else if ret < math.MinInt32 {
        return 0
    }
    return ret
    
}
